lib/atoi/a2i.h

```diff
@@ -0,0 +1,77 @@
+// SPDX-FileCopyrightText: 2023-2025, Alejandro Colomar <alx@kernel.org>
+// SPDX-License-Identifier: BSD-3-Clause
+
+
+#ifndef SHADOW_INCLUDE_LIB_ATOI_A2I_H_
+#define SHADOW_INCLUDE_LIB_ATOI_A2I_H_
+
+
+#include "config.h"
+
+#include <errno.h>
+#include <stddef.h>
+
+#include "atoi/strtoi/strtoi.h"
+#include "atoi/strtoi/strtou_noneg.h"
+#include "typetraits.h"
+
+
+/*
+ * See the manual of these macros in liba2i's documentation:
+ * <http://www.alejandro-colomar.es/share/dist/liba2i/git/HEAD/liba2i-HEAD.pdf>
```

Comment:
1. I can't seem to wget this.
2. A pdf is suboptimal for cross referencing during review. 
3. Can we get a pithy version in line?
4. Wikipedia tells me qchar is not a real type, only for demonstration purposes  :)

Reply:
1)  Yeah, I had to break all links to my server, as I was under persistent DDoS from AI crawlers.
   Use HTTPS on port 80, or HTTP on port 443 (yes, they are swapped).

2)  I could paste the manual page in the commit message.

3)  I'd prefer in commit message.  Or I could put the manual page file in our tree.

4)  Here's the description in the ISO C23 standard:
   <https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3220.pdf#subsubsection.7.26.5.1>
   It is an optionally `const`-qualified `char`.
   It's used so that the third and fourth parameters are equally qualified (either both or none), to have const correctness.

Reply:
The short description is that it is like BSD's strtoi/u(3), but makes sure that the number is stored in a variable of type T (and the bounds are also of type T), and has a const-correct `endp`.

Reply:
Ok